src/engine/readaheadmanager.cpp

```diff
@@ -306,9 +332,11 @@ double ReadAheadManager::getFilePlaypositionFromLog(
 
 mixxx::audio::FramePos ReadAheadManager::getFilePlaypositionFromLog(
         mixxx::audio::FramePos currentPosition,
-        mixxx::audio::FrameDiff_t numConsumedFrames) {
+        mixxx::audio::FrameDiff_t numConsumedFrames,
+        mixxx::audio::ChannelCount channelCount) {
     const double positionSamples =
-            getFilePlaypositionFromLog(currentPosition.toEngineSamplePos(),
-                    numConsumedFrames * mixxx::kEngineChannelCount);
-    return mixxx::audio::FramePos::fromEngineSamplePos(positionSamples);
+            getFilePlaypositionFromLog(currentPosition.toEngineSamplePos(channelCount),
+                    numConsumedFrames * channelCount,
+                    channelCount);
```

Comment:
The term Playposition, is somehow the lagacy stereo assumption form the COs. 
It would be nice to use mixxx::audio::FramePos only in this function.

Reply:
I have done so now (commit is incoming) but that this adding a significant amount of change  (and risk?) to the PR. Happy to move on it's own if you think it makes sense, like for the rubberband multi threading

Reply:
Thank you! Separate PRs are always welcome. Decide yourself if it worth the efforts.

Reply:
I actually could get all the test to pass with my commit, so I will move it to another PR.